sesman/chansrv/chansrv.c

```diff
@@ -861,7 +881,18 @@ process_message_unicode_data(struct stream *s)
 static int
 process_message_unicode_setup(struct stream *s)
 {
-    return xrdp_input_unicode_init();
+    int rv = xrdp_input_unicode_init();
+    if (rv == 0)
+    {
+        // Tell xrdp we can support Unicode input
+        rv = chansrv_advertise_unicode_input(0);
+    }
+    else
+    {
+        // Tell xrdp there's a problem starting the framework
+        chansrv_advertise_unicode_input(2);
```

Comment:
No "rv = " here?

Reply:
At this point `rv` is already non-zero as `xrdp_input_unicode_init()` has failed. I thought it best to pass that error up the call stack within chansrv.